### PR TITLE
Add pom.xml to build gui/

### PR DESCRIPTION
**Pull request by alokmenghrajani**
We need to update the proto version.

Running without the hardware security module:
cd java
mvn install
./gui/target/gui-1.0.0-SNAPSHOT-shaded.jar

Running with a local hardware security module:
same instructions as above, /opt/nfast/java/classes/ should contain nCipherKM.jar.

Running with a remote hardware security module:
scp nCipherKM.jar to /opt/nfast/java/classes/
same instructions as above.

in Intellij:
Until we figure out how to make things work, you may have to manually
adjust the project config. You should probably disable auto-import of
the pom.xml file.

### CODE REVIEW

File: java/gui/pom.xml [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    <artifactId>parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>gui</artifactId>
+
+  <repositories>
+    <repository>
+      <id>square-nexus</id>
+      <url>https://maven.global.square/artifactory/square-public/</url>
```
Comment (mbyczkowski) [inline java/gui/pom.xml]: we want to publish to our public maven repo, correct?

File: java/gui/src/main/java/com/squareup/subzero/actions/InitWallet.java [hunk 2]
```diff
@@ -66,7 +66,7 @@
 
       // Check if we need to create the pub key encryption key (only needs to happen once, on one machine)
       if (subzero.createPubKeyEncryptionKey) {
-        nCipher.createPubKeyEncryptionKey();
+        nCipher.createPubKeyEncryptionKeyTicket();
```
Comment (mbyczkowski) [inline java/gui/src/main/java/com/squareup/subzero/actions/InitWallet.java]: is this intentional?

File: java/gui/pom.xml (hunk 1 above)
Comment (alokmenghrajani) [reply java/gui/pom.xml]: Good catch. This needs to go away.

File: java/gui/src/main/java/com/squareup/subzero/actions/InitWallet.java (hunk 2 above)
Comment (alokmenghrajani) [reply java/gui/src/main/java/com/squareup/subzero/actions/InitWallet.java]: Yes, a rename happened a while ago (commit was squashed) and I missed this spot.

**Pull request merged**